src/aspire/basis/ffb_2d.py

```diff
@@ -193,56 +197,51 @@ def _evaluate_t(self, x):
         n_images = x.shape[0]
 
         # resamping x in a polar Fourier gird using nonuniform discrete Fourier transform
-        pf = nufft(x, 2 * pi * freqs)
-        pf = np.reshape(pf, (n_images, n_r, n_theta))
+        pf = nufft(xp.asarray(x), 2 * pi * freqs)
```

Comment:
This PR adds the capability to do this (provide freqs as cupy) to the nufft wrapper, but you are correct, its not used here for FFB2D. 

I did preload grid in this PR for FLE2D, but the FFB precomp looked like it could use a little more TLC than just GPU changes.  I will make a follow up issue.

Reply:
https://github.com/ComputationalCryoEM/ASPIRE-Python/issues/1158

Reply:
> This PR adds the capability to do this (provide freqs as cupy) to the nufft wrapper, but you are correct, its not used here for FFB2D.

Ok I didn't see this. Cool!